src/AnnouncementIndex.coffee

```diff
@@ -20,21 +22,29 @@ class AnnouncementIndex
       @discoveryNotifier.notifyWatchers(update)
 
   removeAnnouncements:(ids=[])->
-    @announcements = _.omit(@announcements, ids)
+    @_announcements = _.omit(@_announcements, ids)
 
   addAnnouncements:(announcements=[])->
-    @announcements = _.extend(
-      @announcements,
-      _.indexBy(announcements, "announcementId")
-    )
+    for announce in announcements
+      aId = announce.announcementId
+      if aId in @_announcements
+        # here we wont override an existing announcement if we already have one from our region and the
+        # new announcement is from another region
```

Comment:
Ok I was 50/50 on which way to go with that.  I'll do some exploring on that.  The issue I see is having the same announcementId from different regions.